scripts/jit/log_extract.py

```diff
@@ -0,0 +1,148 @@
+from contextlib import contextmanager
+from typing import Any, List, Tuple
+import argparse
+import torch
+
+
+def extract_ir(filename: str) -> List[str]:
+    BEGIN = "<GRAPH_EXPORT>"
+    END = "</GRAPH_EXPORT>"
+    pfx = None
+    current = ""
+    graphs = []
+    with open(filename, "r") as f:
+        for line in f.readlines():
+            begin_loc = line.find(BEGIN)
+            if begin_loc != -1:
+                pfx = line[:begin_loc]
+                continue
+
+            end_loc = line.find(END)
+            if end_loc != -1:
+                graphs.append(current)
+                current=""
+                pfx = None
+                continue
+
+            if not pfx:
+                continue
+
+            if len(line) < len(pfx) or line[:len(pfx)] != pfx:
+                raise RuntimeError("Expected prefix of '" + pfx + "'")
+
+            current += line[len(pfx):]
+
+    return graphs
+
+
+def load_graph_and_inputs(ir: str) -> Tuple[Any, List[Any]]:
+    graph = torch._C.parse_ir(ir)
+    graph.makeMultiOutputIntoTuple()
+    inputs = []
+    for inp in graph.inputs():
+        if isinstance(inp.type(), torch._C.FloatType):
+            inputs.append(.5)
+        else:
+            inputs.append(torch._C._representative_tensor(inp.type()))
+
+    func = torch._C._create_function_from_graph("forward", graph)
+    torch._C._jit_pass_erase_shape_information(func.graph)
+    return (func, inputs)
+
+
+def run_test(ir, inputs, *, warmup_runs=20, test_runs=20) -> float:
```

Comment:
warmup_runs == 20 is probably a bit too much, especially if we're not running gradients. Same with 20 test runs.